staging/src/k8s.io/code-generator/cmd/lister-gen/generators/lister.go

```diff
@@ -286,45 +283,21 @@ type $.type|public$Lister interface {
 var typeListerStruct = `
 // $.type|private$Lister implements the $.type|public$Lister interface.
 type $.type|private$Lister struct {
```

Comment:
Can we document why we choose to embed:

```
type fooLister struct {
    listers.ResourceIndexer[foo]
}
```

vs. alias:

```
type fooLister = listers.ResourceIndexer[foo]
```

?  If there are tradeoffs, can we put them in a comment here?

Reply:
I mentioned this in the commit message and PR description, but I’ve added a comment above both structs.